### PR TITLE
add GetDialogDismissTask method

### PR DESCRIPTION
**Pull request by sleushunou**

### CODE REVIEW

File: Softeq.XToolkit.WhiteLabel.Droid/Providers/ContextProvider.cs [hunk 1]
```diff
@@ -1,22 +1,48 @@
...
         public Context AppContext => Xamarin.Essentials.Platform.AppContext;
+
+        private void OnPlatformActivityStateChanged(object sender, Xamarin.Essentials.ActivityStateChangedEventArgs e)
+        {
+            if (_activityRef.Target != CurrentActivity)
+            {
+                CurrentActivityChanged?.Invoke(this, EventArgs.Empty);
+            }
+
+            _activityRef = WeakReferenceEx.Create(CurrentActivity);
```
Comment (wcoder) [inline Softeq.XToolkit.WhiteLabel.Droid/Providers/ContextProvider.cs:45]: what about creating ref in one place, here?
and skip cases when _activityRef = null or CurrentActivity = null

File: Softeq.XToolkit.WhiteLabel.Droid/Providers/ContextProvider.cs [hunk 2]
```diff
@@ -1,22 +1,48 @@
...
     }
 
     public class EssentialsContextProvider : IContextProvider
     {
+        private WeakReferenceEx<Activity> _activityRef;
+
+        public EssentialsContextProvider()
+        {
+            _activityRef = WeakReferenceEx.Create(CurrentActivity);
+            Xamarin.Essentials.Platform.ActivityStateChanged += OnPlatformActivityStateChanged;
```
Comment (TimofeyBurak) [inline Softeq.XToolkit.WhiteLabel.Droid/Providers/ContextProvider.cs:29]: I'd say we could also expose original `ActivityStateChanged` event here, and let the user of the API decide how to use it.. 
BTW, this way we could implement method `WaitForCurrentActivityChangeAsync()` as an extension to IContextProvider interface

File: Softeq.XToolkit.WhiteLabel.Droid/Providers/ContextProvider.cs [hunk 3]
```diff
@@ -1,22 +1,49 @@
...
 
+    public class ActivityStateChangedEventArgs : EventArgs
+    {
+        internal ActivityStateChangedEventArgs(Activity activity, ActivityState state)
+        {
+            State = state;
+            Activity = activity;
+        }
+
+        public ActivityState State { get; }
```
Comment (wcoder) [inline Softeq.XToolkit.WhiteLabel.Droid/Providers/ContextProvider.cs:30]: `ActivityState` is enum from Essentials

Comment (wcoder) [reply Softeq.XToolkit.WhiteLabel.Droid/Providers/ContextProvider.cs:30]: we can use `Lifecycle.State` from AndroidX

**Review by TimofeyBurak — APPROVED** [review]

**Pull request merged**